src/components/subscribe/mailchimp-subscribe.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { SubscribeForm } from "./subscribe-form";
+import { sendSubscriptionRequest } from "./mailchimp.service";
+
+export const MailchimpSubscribe = ({ url }) => {
+  const [isSending, setIsSending] = useState();
+  const [isSuccess, setIsSuccess] = useState();
+  const [errorMessage, setErrorMessage] = useState();
+
+  const subscribe = async ({ email }) => {
+    setIsSending(true);
+    try {
+      await sendSubscriptionRequest(url, email);
+      setIsSending(false);
+      setIsSuccess(true);
+    } catch (errorMsg) {
+      setErrorMessage(errorMsg);
+    }
```

Comment:
We should toggle `isSending` in a finally block so we're covered when the request fails as well.
```suggestion
    try {
      await sendSubscriptionRequest(url, email);
      setIsSuccess(true);
    } catch (errorMsg) {
      setErrorMessage(errorMsg);
    } finally {
      setIsSending(false); 
    }
```

Reply:
Should we also `setIsSuccess(false);` in the catch block?

Reply:
success will become true only if the request succeeds. 
So no need IMHO.